data_models/data_refinery_models/models/jobs.py

```diff
@@ -1,73 +1,64 @@
+from django.db import transaction
 from django.db import models
 from data_refinery_models.models.base_models import TimeTrackedModel
 from data_refinery_models.models.batches import Batch
 
 
-class ProcessorJob(TimeTrackedModel):
-    """Records information about running a processor."""
+class WorkerJob(TimeTrackedModel):
+    """Base model with auto created_at and updated_at fields."""
+
+    class Meta:
+        abstract = True
 
+    batches = models.ManyToManyField(Batch)
     start_time = models.DateTimeField(null=True)
     end_time = models.DateTimeField(null=True)
     success = models.NullBooleanField(null=True)
 
-    # This field will contain an enumerated value specifying which processor
-    # pipeline was applied during the processor job.
-    pipeline_applied = models.CharField(max_length=256)
-
-    # This field represents how many times this job has been retried. It starts
-    # at 0 and each time the job has to be retried it will be incremented.
-    # At some point there will probably be some code like:
-    # if job.num_retries >= 3:
-    #     # do a bunch of logging
-    # else:
-    #     # retry the job
+    # This field represents how many times this job has been
+    # retried. It starts at 0 and each time the job has to be retried
+    # it will be incremented.
     num_retries = models.IntegerField(default=0)
 
-    # This point of this field is to identify which worker ran the job.
-    # A few fields may actually be required or something other than just an id.
-    worker_id = models.CharField(max_length=256, null=True)
+    # This field indicates whether or not this job has been retried
+    # already or not.
+    retried = models.BooleanField(default=False)
 
-    class Meta:
-        db_table = "processor_jobs"
+    # This point of this field is to identify which worker ran the
+    # job. A few fields may actually be required or something other
+    # than just an id.
+    worker_id = models.CharField(max_length=256, null=True)
 
+    @classmethod
+    @transaction.atomic
+    def create_job_and_relationships(cls, *args, **kwargs):
+        """Inits and saves a job and its relationships to its batches."""
```

Comment:
Hmm... I noticed that this class method needs some arguments and inside the body it plays with the arguments. Since this method has been called a lot of times by other modules, it will be great to add some detailson how it should be called.